app/Http/Controllers/Web/ProductController.php

```diff
@@ -0,0 +1,112 @@
+<?php
+
+namespace App\Http\Controllers\Web;
+
+use App\Http\Controllers\Controller;
+use App\Models\Product;
+use App\Models\Category;
+use App\Repositories\Contracts\CategoryInterface;
+use App\Repositories\Contracts\ProductInterface;
+use App\Services\ProductService;
+use Illuminate\Http\Request;
+
+class ProductController extends Controller
+{
+    protected $productRepository, $categoryRepository;
+    protected $productService;
+
+    public function __construct(ProductInterface $productRepository, CategoryInterface $categoryRepository, ProductService $productService)
```

Comment:
đoạn này xuống dòng cho anh nhé
```suggestion
    public function __construct(
        ProductInterface $productRepository, 
        CategoryInterface $categoryRepository, 
        ProductService $productService)
```